src/main/java/dev/morling/onebrc/CalculateAverage_ericxiao.java

```diff
@@ -300,32 +278,35 @@ public static void main(String[] args) throws Exception {
             finally {
                 executorService.shutdown();
                 // fileChannel.close();
-                Map<ProcessFileMap.KeySlice, int[]> mapA = results.getFirst();
+                Station[] mapA = results.getFirst();
                 for (int i = 1; i < numThreads; ++i) {
-                    results.get(i).forEach((station, stationMeasurements) -> {
-                        if (mapA.containsKey(station)) {
-                            int[] measurements = mapA.get(station);
-                            measurements[0] = Math.min(measurements[0], stationMeasurements[0]);
-                            measurements[1] = Math.max(measurements[1], stationMeasurements[1]);
-                            measurements[2] = measurements[2] + stationMeasurements[2];
-                            measurements[3] = measurements[3] + stationMeasurements[3];
+                    Station[] currMap = results.get(i);
+                    for (int j = 0; j < 10000; j++) {
+                        if (currMap[j] != null) {
+                            if (mapA[j] != null) {
+                                mapA[j].min = Math.min(mapA[j].min, currMap[j].min);
+                                mapA[j].max = Math.max(mapA[j].max, currMap[j].max);
+                                mapA[j].sum += currMap[j].sum;
+                                mapA[j].count += currMap[j].count;
+                            }
+                            else {
+                                mapA[j] = currMap[j];
+                            }
                         }
-                        else {
-                            mapA.put(station, stationMeasurements);
-                        }
-                    });
+                    }
                 }
                 // print key and values
                 int counter = 1;
                 System.out.print("{");
-                for (Map.Entry<ProcessFileMap.KeySlice, int[]> entry : mapA.entrySet()) {
-                    int[] measurements = entry.getValue();
-                    double mean = (double) measurements[2] / (double) measurements[3];
-                    System.out.print(entry.getKey().key + "=" + (measurements[0] / 10.0) + "/"
-                            + (Math.round(mean) / 10.0) + "/"
-                            + (measurements[1]) / 10.0);
-                    if (counter++ < mapA.size())
-                        System.out.print(", ");
+                for (Station entry : mapA) {
+                    if (entry != null) {
+                        double mean = (double) entry.sum / (double) entry.count;
+                        System.out.print(entry.station + "=" + (entry.min / 10.0) + "/"
+                                + (Math.round(mean) / 10.0) + "/"
+                                + (entry.max / 10.0));
+                        if (counter++ < 403) // TODO: hard coding for now.
```

Comment:
@damnMeddlingKid, other remaining comment - since our hashmap has empty elements and we don't know the key count beforehand, we need to make sure the printing is handled correctly.